### PR TITLE
Added test for general algorithm and speedup for toeplitz case

### LINKED ISSUE #1: Unit tests
# Description

Implement unit tests to improve code coverage.
- [x] general schur algorithm
- [x] qrtoeplitz
- [x] lstoeplits
- [x] blocktoeplitz
- [x] hyperbolic rotations
# Need

For confidence.
# References / Links

N/A.
# Proposal

N/A.

### PR DESCRIPTION
**Pull request by neveritt**
For toeplitz case, the downshift is now implemented as a shift and
not as a multiplication by a sparse matrix. Fixes #1

### CODE REVIEW

**Comment by neveritt** [comment]
@aytekinar can you please review the pull request

File: src/blocktoeplitz.jl [hunk 1]
```diff
@@ -52,8 +58,10 @@ function size(A::BlockToeplitz, dim::Int)
     throw(DomainError())
   end
 end
+ndims(A::BlockToeplitz)              = 2
```
Comment (aytekinar) [inline src/blocktoeplitz.jl]: `ndims` not needed

File: src/blocktoeplitz.jl [hunk 2]
```diff
@@ -52,8 +58,10 @@ function size(A::BlockToeplitz, dim::Int)
...
+size(A::BlockToeplitz)               = (size(A.vc,1), size(A.vr,2))
```
Comment (aytekinar) [inline src/blocktoeplitz.jl]: `size(A::BlockToeplitz) = (m,n)` maybe?

File: src/blocktoeplitz.jl [hunk 3]
```diff
@@ -52,8 +58,10 @@ function size(A::BlockToeplitz, dim::Int)
...
+size(A::BlockToeplitz, dims::Int...) = map(x-> size(A, x), dims)
```
Comment (aytekinar) [inline src/blocktoeplitz.jl]:
``` julia
function size(A::BlockToeplitz, dim::Int)
  @assert dim > 0 "size: dim must be positive"
  dim == 1 ? m : dim == 2 ? n : 1
end
```

**Review by aytekinar — CHANGES_REQUESTED** [review]
some small polishing

File: src/blocktoeplitz.jl (hunk 2 above)
Comment (neveritt) [reply src/blocktoeplitz.jl]: no, m and n are the number of blocks in each dimension

File: src/blocktoeplitz.jl (hunk 1 above)
Comment (neveritt) [reply src/blocktoeplitz.jl]: thx

File: src/blocktoeplitz.jl (hunk 3 above)
Comment (neveritt) [reply src/blocktoeplitz.jl]: thx

**Review by aytekinar — APPROVED** [review]

**Pull request merged**